mono/metadata/locales.c

```diff
@@ -530,22 +532,40 @@ get_current_locale_name (void)
 	return ret;
 }
 
-MonoString*
-ves_icall_System_Globalization_CultureInfo_get_current_locale_name (void)
+static void
+culture_info_get_current_locale_name (MonoStringHandle ret, MonoError *error)
```

Comment:
I'm probably missing something, here's what we can do for returning managed objects from functions:

0. Always return a global GC handle.
1. Pass a caller-allocated local handle to the callee.  Callee assigns a value to it.
2. Like (1) but use TLS to point at the caller-allocated handle so we don't have to pass anything explicitly to the callee.
3. Give the callee an API to allocate a handle in the caller's handle arena.
4. For the runtime we know we're in RUNNING state, so just return a raw pointer and have a convention that the caller must adopt it into a local handle right away.  Do something else for the API.

I kind of like 3, but what I have now (1) is simpler to explain.